goodtablesio/plugins/s3/utils.py

```diff
@@ -0,0 +1,345 @@
+import json
+import logging
+
+import boto3
+import botocore
+
+from goodtablesio import settings, exceptions
+
+
+log = logging.getLogger(__name__)
+
+
+class S3Exception(exceptions.GoodtablesioException):
+    pass
+
+
+# TODO: check connection :invalid access key
+
+class S3Client(object):
+
+    def __init__(self, access_key_id, secret_access_key):
+        """
+        Initializes a client for a user provided S3 bucket
+
+        Arguments:
+            access_key_id (str): The Access Key Id part of the credentials
+            secret_access_key (str): The Secret Access Key part of the
+                credentials
+
+        """
+        self.client = boto3.client(
+            's3',
+            aws_access_key_id=access_key_id,
+            aws_secret_access_key=secret_access_key)
+
+    def _notification_id_for_bucket(self, bucket_name):
```

Comment:
It's style related but I suppose it's easier to understand class API if public methods goes first then private methods.